### PR TITLE
fix: SearchView data persists on rotation

### LINKED ISSUE #862: SearchView closes and data lost on rotation in search fragment.
**Describe the bug**

On rotation in search fragment while searchview is open, focus and data lost on rotation

**To Reproduce**

1. Go to the search fragment.
2. Open search option and enter text 
3. Rotate device.

**Expected behavior**

SearchView should remain in focus without loss of data

**Logs**

**Screenshots**

search_rotation

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |One Plus 5|
|Android Version|Android Pie 9.0.0  |

**Additional context**

@iamareebjamal @nikit19 Your views?

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by kshitijsuri90**
Fixes #862 

Changes:
Added MutableLiveData in SearchViewModel and setting the value to that through onQueryChange listener. A logic is added to check whether the MutableLiveData is set or not.

Screenshots for the change:
search_
@nikit19 @iamareebjamal Do review and suggest changes if any.

### CODE REVIEW

**Comment by kshitijsuri90** [comment]
Is this correct @iamareebjamal @nikit19 ?

**Comment by kshitijsuri90** [comment]
Is this the correct way to do this @iamareebjamal?

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt [hunk 1]
```diff
@@ -90,6 +92,11 @@ class SearchFragment : Fragment() {
         val thisActivity = activity
         if (thisActivity is MainActivity) searchView = SearchView(thisActivity.supportActionBar?.themedContext)
         MenuItemCompat.setActionView(searchItem, searchView)
+        searchViewModel.queryText.observe(this, Observer {
+            searchItem.expandActionView()
+            searchView.setQuery(searchViewModel.queryText.value, false)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: No

Comment (kshitijsuri90) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: What exact changes are needed to made her sir? @iamareebjamal

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: Can you tell me what is the use of Observer of LiveData. By looking at this code, it is clear that you don't know how livedata works

**Comment by kshitijsuri90** [comment]
@liveHarshit Do review.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt [hunk 2]
```diff
@@ -89,6 +91,11 @@ class SearchFragment : Fragment() {
         val thisActivity = activity
         if (thisActivity is MainActivity) searchView = SearchView(thisActivity.supportActionBar?.themedContext)
         searchItem.actionView = searchView
+        searchViewModel.queryText.observe(this, Observer {
+            searchItem.expandActionView()
+            searchView.setQuery(it, false)
+            searchView.maxWidth = Resources.getSystem().displayMetrics.widthPixels
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: Why this line?

**Comment by iamareebjamal** [comment]
Squash your commits

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt (hunk 2 above)
Comment (kshitijsuri90) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: On rotation the search bar is not occupying the whole screen size

**Pull request closed without merging**